sentinel/server.js

```diff
@@ -17,34 +19,20 @@ process.on('unhandledRejection', reason => {
   console.error(reason);
 });
 
-const MIN_MAJOR = 8;
-const nodeVersionCheck = () => {
-  try {
-    const [major, minor, patch] = process.versions.node.split('.').map(Number);
-    if (major < MIN_MAJOR) {
-      throw new Error(`Node version ${major}.${minor}.${patch} is not supported, minimum is ${MIN_MAJOR}.0.0`);
-    }
-    console.log(`Node Version: ${major}.${minor}.${patch}`);
-  } catch (err) {
-    console.error('Fatal error intialising medic-sentinel');
-    console.log(err);
-    process.exit(1);
-  }
-};
-
-nodeVersionCheck();
-
-config.init()
+serverChecks.check(db)
   .then(() => {
-    if (!loglevel) {
-      logger.transports.Console.level = config.get('loglevel');
-      logger.debug('loglevel is %s.', logger.transports.Console.level);
-    }
-    require('./src/schedule').checkSchedule();
-    logger.info('startup complete.');
-  })
-  .catch(err => {
-    console.error('Fatal error intialising medic-sentinel');
-    console.log(err);
-    process.exit(1);
+    config.init()
+      .then(() => {
+        if (!loglevel) {
+          logger.transports.Console.level = config.get('loglevel');
+          logger.debug('loglevel is %s.', logger.transports.Console.level);
+        }
+        require('./src/schedule').checkSchedule();
+        logger.info('startup complete.');
+      })
+      .catch(err => {
+        console.error('Fatal error intialising medic-sentinel');
+        console.log(err);
+        process.exit(1);
+      });
```

Comment:
If I'm reading this right then we won't log anything if the `serverChecks` fails because the `catch` is inside the `then`. Instead, move the `catch` to the top level and return from `config.init()`, eg:

```js
serverChecks.check(db)
  .then(() => config.init())
  .then(() => {
    if (!loglevel) {
      logger.transports.Console.level = config.get('loglevel');
      logger.debug('loglevel is %s.', logger.transports.Console.level);
    }
    require('./src/schedule').checkSchedule();
    logger.info('startup complete.');
  })
  .catch(err => {
    console.error('Fatal error intialising medic-sentinel');
    console.log(err);
    process.exit(1);
  });
```